tests/php/Collaboration/Collaborators/RoomPluginTest.php

```diff
@@ -0,0 +1,255 @@
+<?php
+declare(strict_types=1);
+
+/**
+ *
+ * @copyright Copyright (c) 2018, Daniel Calviño Sánchez (danxuliu@gmail.com)
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Spreed\Tests\php\Collaboration\Collaborators;
+
+use OCA\Spreed\Collaboration\Collaborators\RoomPlugin;
+use OCA\Spreed\Manager;
+use OCA\Spreed\Room;
+use OCP\Collaboration\Collaborators\ISearchResult;
+use OCP\Collaboration\Collaborators\SearchResultType;
+use OCP\IDBConnection;
+use OCP\IUser;
+use OCP\IUserSession;
+use OCP\Security\IHasher;
+use OCP\Security\ISecureRandom;
+use OCP\Share;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class RoomPluginTest extends \Test\TestCase {
+
+	/** @var Manager */
+	protected $manager;
+
+	/** @var IUserSession */
+	protected $userSession;
+
+	/** @var IUser */
+	protected $user;
+
+	/** @var ISearchResult */
+	protected $searchResult;
+
+	/** @var RoomPlugin */
+	protected $plugin;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->manager = $this->createMock(Manager::class);
+
+		$this->user = $this->createMock(IUser::class);
+		$this->user->expects($this->any())
+			->method('getUID')
+			->willReturn('user0');
+		$this->userSession = $this->createMock(IUserSession::class);
+		$this->userSession->expects($this->any())
+			->method('getUser')
+			->willReturn($this->user);
+
+		$this->searchResult = $this->createMock(ISearchResult::class);
+
+		$this->plugin = new RoomPlugin($this->manager, $this->userSession);
+	}
+
+	private function newRoom(string $id, int $type, string $token, string $name): Room {
+		$manager = $this->createMock(Manager::class);
+		$db = $this->createMock(IDBConnection::class);
+		$secureRandom = $this->createMock(ISecureRandom::class);
+		$dispatcher = $this->createMock(EventDispatcherInterface::class);
+		$hasher = $this->createMock(IHasher::class);
+		$password = '';
+		$activeGuests = 0;
+
+		return new Room($manager, $db, $secureRandom, $dispatcher, $hasher, $id, $type, $token, $name, $password, $activeGuests);
```

Comment:
could also do `$this->createMock(Room::class);` and then mock the methods you actually test?

Reply:
Good point, done; I kept the room type as a parameter even if it is not actually used because it looks like something that should be included in the result, so less typing in the tests for the future :-P